utils/lite-components-testkit/src/main/scala/pl/touk/nussknacker/engine/lite/util/test/RequestResponseTestScenarioRunner.scala

```diff
@@ -16,16 +19,30 @@ import scala.reflect.ClassTag
 
 class RequestResponseTestScenarioRunner(val components: List[ComponentDefinition], val config: Config) extends ClassBaseTestScenarioRunner with PatientScalaFutures {
 
-  override def runWithData[I:ClassTag, R](scenario: EspProcess, data: List[I]): List[R] = {
+  override def runWithData[I:ClassTag, R](scenario: EspProcess, data: List[I]): RunnerResult[R] = {
     val (modelData, runId) = ModelWithTestComponents.prepareModelWithTestComponents(config, components)
     import pl.touk.nussknacker.engine.requestresponse.FutureBasedRequestResponseScenarioInterpreter._
-    try {
-      val interpreter = RequestResponseInterpreter[Future](scenario,
-        ProcessVersion.empty, LiteEngineRuntimeContextPreparer.noOp, modelData, Nil, null, null).getOrElse(throw new IllegalArgumentException(""))
-      Future.sequence(data.map(interpreter.invokeToOutput)).futureValue.map(_.asInstanceOf[R])
-    } finally {
-      TestComponentsHolder.clean(runId)
-    }
+
+    val result = RequestResponseInterpreter[Future](
+      scenario, ProcessVersion.empty,
+      LiteEngineRuntimeContextPreparer.noOp, modelData,
+      Nil, null, null
+    ).map(interpreter => {
```

Comment:
notation + you can do both collecting by using `foldLeft`. This solution will be less error prone because this two blocks can desynchronize now.